src/expressions/block-expr.md

```diff
@@ -80,6 +80,64 @@ fn move_by_block_expression() {
 }
 ```
 
+## `async` blocks
+
+> **<sup>Syntax</sup>**\
+> _AsyncBlockExpression_ :\
+> &nbsp;&nbsp; `async` `move`<sup>?</sup> _BlockExpression_
+
+An *async block* is a variant of a block expression which evalutes to
+a *future*. The final expression of the block (if present) determines
+the result value of the future.
+
+Executing an async block is thus similar to executing a closure
+expression: its immediate effect is to produce and return a struct.
+Whereas closures return a struct that implements one or more of the
+`std::ops::Fn` traits, however, the type returned for an async block
+implements the `std::future::Future` trait. The actual data format for
+this type is undefined, but it is roughly equivalent to an enum with
+one variant per `await` point (such that we store the data needed to
+resume from that point).
+
+Async blocks capture variables from their environment in a way similar
+to closures. That is, when the async block is first executed, any
+variables referenced in the block will be "captured" into the
+resulting future. In the absence of the `move` keyword, async blocks
+will capture in the weakest way possible, depending on how the
+variable is used (just as closures do). That is, if the variable is
+only used to produce shared references, or is taken by value but is of
```

Comment:
Link to shared references (https://doc.rust-lang.org/nightly/reference/types/pointer.html#shared-references-)